sql/analyzer/unnest_insubqueries.go

```diff
@@ -122,30 +127,29 @@ func transformJoinApply(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.S
 				//     normalized to match changes to (2).
 				subq := m.r
 
-				name := fmt.Sprintf("scalarSubq%d", applyId)
-				applyId++
-
-				sch := subq.Query.Schema()
-				var rightF sql.Expression
-				if len(sch) == 1 {
-					subqCol := subq.Query.Schema()[0]
-					rightF = expression.NewGetFieldWithTable(len(scope.Schema()), subqCol.Type, subqCol.DatabaseSource, name, subqCol.Name, subqCol.Nullable)
-				} else {
-					tup := make(expression.Tuple, len(sch))
-					for i, c := range sch {
-						tup[i] = expression.NewGetFieldWithTable(len(scope.Schema())+i, c.Type, c.DatabaseSource, name, c.Name, c.Nullable)
+				if aliases == nil {
```

Comment:
I think you can move this block out of the for loop, and remove the `nil` check.

Reply:
i want to lazily collect aliases